event.go

```diff
@@ -0,0 +1,376 @@
+}
+
+// DeviceStateChanged events indicate that a device state has changed
+type DeviceStateChanged struct {
+	EventData
+	Device_state DeviceState `json:"device_state"`
+}
+
+// Dial events indicate the dialing state for a channel has changed
+type Dial struct {
+	EventData
+	Caller     Channel `json:"caller,omitempty"`     // Dialing channel (if not system-originated)
+	Dialstatus string  `json:"dialstatus"`           // Present status of the dial attempt
+	Dialstring string  `json:"dialstring,omitempty"` // The string describing the dial
+	Forward    string  `json:"forward,omitempty"`    // If present, indicates the forwarding target TODO: what is the format of this?
+	Forwarded  Channel `json:"forwarded,omitempty"`  // If present, channel of forwarding target
+	Peer       Channel `json:"peer"`                 // Dialed channel
+}
+
+// EndpointStateChange events indicate that an endpoint has changed state
+type EndpointStateChange struct {
+	EventData
+	Endpoint Endpoint `json:"endpoint"`
+}
+
+/* --------- Peer Events ---------------- */
+
+// PeerStatusChange events indicate that a peer has changed state
+type PeerStatusChange struct {
+	EventData
+	Endpoint Endpoint `json:"endpoint"`
+	Peer     Peer     `json:"peer"`
+}
+
+/* --------- Playback Events ---------------- */
+
+// PlaybackFinished events indicate that a media playback operation has completed
+type PlaybackFinished struct {
+	EventData
+	Playback Playback `json:"playback"`
+}
+
+// PlaybackStarted events indicate that a media playback operation has begun
+type PlaybackStarted struct {
+	EventData
+	Playback Playback `json:"playback"`
+}
+
+/* --------- Recording Events ---------------- */
+
+/* These three are commented out until LiveRecording is taken care of
+// RecordingFailed events indicate that a recording operation request has failed to complete
+type RecordingFailed struct {
+	EventData
+	Recording LiveRecording `json:"recording"`
+}
+
+// RecordingFinished events indicate that a recording operation has completed
+type RecordingFinished struct {
+	EventData
+	Recording LiveRecording `json:"recording"`
+}
+
+// RecordingStarted events indicate that a recording operation has begun
+type RecordingStarted struct {
+	EventData
+	Recording LiveRecording `json:"recording"`
+}
+*/
+
+/* --------- Stasis Events ---------------- */
+
+// StasisEnd events indicate that a channel has left the Stasis (Ari) application
+type StasisEnd struct {
+	EventData
+	Args []string `json:"args"`
+	ChannelData
+}
+
+// StasisStart events indicate a channel has entered the Stasis (Ari) application
+type StasisStart struct {
+	EventData
+	Args []string `json:"args"`
+	ChannelData
+	ReplaceChannel Channel `json:"replacechannel,omitempty"` // TODO: find out what this is
+}
+
+// R2 - SKIPPED FOR NOW
+// GetChannel returns the channel for whom the StasisStart event occurred,
+// optionally attaching a copy of the ARI client
+/*
+func (s *StasisStart) GetChannel(c *Client) Channel {
```

Comment:
You want to implement this in terms of ari.ChannelHandle and ari.Channel interface.